recipes/itk/all/conanfile.py

```diff
@@ -242,6 +233,29 @@ def _patch_sources(self):
                         os.path.join(self.source_folder, "Modules", "ThirdParty", "VNL", "src", "vxl", "config", "cmake", "config", "VXLIntrospectionConfig.cmake"),
                         "-DCMAKE_CXX_FLAGS:STRING=${CMAKE_CXX_FLAGS}",
                         "-DCMAKE_POLICY_DEFAULT_CMP0091=NEW -DCMAKE_CXX_FLAGS:STRING=${CMAKE_CXX_FLAGS}")
+        # Ensure that new versions don't introduce any vendored libs by accident
+        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"), "include(ExternalProject)", "")
+        # Unvendor ZLIB
+        for path in [
+            os.path.join(self.source_folder, "Modules", "IO", "GIPL", "src", "itkGiplImageIO.cxx"),
+            os.path.join(self.source_folder, "Modules", "IO", "PhilipsREC", "src", "itkPhilipsRECImageIO.cxx"),
+            os.path.join(self.source_folder, "Modules", "Nonunit", "Review", "src", "itkVoxBoCUBImageIO.cxx"),
+            os.path.join(self.source_folder, "Modules", "ThirdParty", "GDCM", "src", "gdcm", "Utilities", "gdcm_zlib.h"),
+            os.path.join(self.source_folder, "Modules", "ThirdParty", "GIFTI", "src", "gifticlib", "gifti_io.h"),
+            os.path.join(self.source_folder, "Modules", "ThirdParty", "NIFTI", "src", "nifti", "znzlib", "znzlib.h"),
+            os.path.join(self.source_folder, "Modules", "ThirdParty", "MetaIO", "src", "MetaIO", "src", "localMetaConfiguration.h"),
+            os.path.join(self.source_folder, "Modules", "ThirdParty", "MINC", "src", "libminc", "nifti", "znzlib.h"),
+            os.path.join(self.source_folder, "Modules", "ThirdParty", "NrrdIO", "src", "NrrdIO", "unteem.pl"),
+            os.path.join(self.source_folder, "Modules", "ThirdParty", "NrrdIO", "src", "NrrdIO", "privateNrrd.h"),
+        ]:
+            replace_in_file(self, path, "itk_zlib.h", "zlib.h")
+        # Truncate some third-party modules that are provided by conan_cmake_project_include.cmake
+        for pkg in ["DCMTK", "DoubleConversion", "GDCM", "Expat", "HDF5", "JPEG", "PNG", "TIFF", "ZLIB"]:
```

Comment:
```suggestion
        # Truncate some third-party modules that are provided by conan_cmake_project_include.cmake
        for pkg in ["DCMTK", "DoubleConversion", "GDCM", "Expat", "HDF5", "JPEG", "PNG", "TIFF"]:
```

Not needed to manipulate zlib, all files have a ifdef to detect what should be included. Those that are using itk_zlib.h will point to zlib as well due cmake configuration.

- [InsightSoftwareConsortium/ITK@v5.2.1/Modules/ThirdParty/ZLIB/src/itk_zlib.h.in](https://github.com/InsightSoftwareConsortium/ITK/blob/v5.2.1/Modules/ThirdParty/ZLIB/src/itk_zlib.h.in)
- [InsightSoftwareConsortium/ITK@9eb1359/Modules/IO/GIPL/src/itkGiplImageIO.cxx#L22](https://github.com/InsightSoftwareConsortium/ITK/blob/9eb13596cef6009c026b2ee837ea52ee05d00f11/Modules/IO/GIPL/src/itkGiplImageIO.cxx#L22)